File: VB/Console/Complex.vb
class Complex
	dim real as single
	dim imag as single
	
	sub new()
	
	end sub
	sub new(r as single, i as single)
		real = r
		imag = i
	end sub
	function add(x as Complex) as Complex
		dim t as new Complex()
		t.real = real + x.real
		t.imag = real + x.imag
		return t
	end function
	sub display()
		Console.Writeline("real: {0}, imaginary: {1}", real, imag)
	end sub
end class

module ComplexTest
	sub main
		dim c1 as new Complex(10.5, 5.2)
		dim c2 as new Complex(20.5, 7.2)
		dim c3 as new Complex()
		c3 = c1.add(c2)
		c3.display()
	end sub
end module

File: CPP/Complex.cpp
#include<iostream>

using namespace std;

class complex
{
	float real;
	float imag;
	
public:
	complex() {}
	complex(float r, float i)
	{
		real = r;
		imag = i;	
	}
	
	void display()
	{
		cout<<"real:"<<real<<endl;
		cout<<"imag:"<<imag<<endl;
	}
	
	complex operator+(complex x);
	complex operator+(int x);
	friend complex operator+(int x, complex c);
};

complex complex::operator+(complex x)
{
	complex t;
	
	t.real = this->real + x.real;
	t.imag = this->imag + x.imag;
	
	return t;
}

complex complex::operator+(int x)
{
	complex t;
	
	t.real = this->real + x;
	t.imag = this->imag + x;
	
	return t;
}

complex operator+(int x, complex c)
{
	complex t;
	
	t.real = c.real + x;
	t.imag = c.imag + x;
	
	return t;
}

int main()
{
	complex c1(5.3, 8.9), c2(9.9, 3.6), c3, c4, c5, c6;
	
	c3 = c1 + c2;
	
	c4 = c2 + c1;

	c5 = c1 + 2;
	
	c6 = 2 + c1;
		
	//c3 = c1.sum(c2);
	
	c3.display();
	c4.display();
	c5.display();
	c6.display();
	
	return 0;
}
